gravitee-apim-console-webui/src/portal/homepage/homepage.component.ts

```diff
@@ -30,14 +36,28 @@ import { GioPermissionService } from '../../shared/components/gio-permission/gio
 export class HomepageComponent {
   contentControl = new FormControl('');
 
-  private readonly originalValue = 'Homepage content';
-  private readonly content = signal<string>(this.originalValue);
-  private readonly canUpdate = signal(inject(GioPermissionService).hasAnyMatching(['environment-documentation-u']));
+  private readonly snackbarService = inject(SnackBarService);
+  private readonly portalPagesService = inject(PortalPagesService);
+  private readonly gioPermissionService = inject(GioPermissionService);
+
+  private readonly canUpdate = signal(this.gioPermissionService.hasAnyMatching(['environment-documentation-u']));
+  private readonly portalHomepage: Signal<PortalPageWithDetails> = toSignal(this.getPortalHomepage(), {
+    initialValue: { content: '' } as PortalPageWithDetails,
+  });
 
   constructor() {
     effect(() => {
-      this.contentControl.setValue(this.content());
+      this.contentControl.setValue(this.portalHomepage().content);
       !this.canUpdate() && this.contentControl.disable();
     });
   }
+
+  private getPortalHomepage(): Observable<PortalPageWithDetails> {
+    return this.portalPagesService.getHomepage().pipe(
+      catchError(() => {
+        this.snackbarService.error('An error occurred while loading the homepage');
+        return of();
```

Comment:
WDYT about 
```suggestion
        return of(undefined);
```
Returning of(undefined) explicitly sets the signal's value to undefined, making the error state clear and easy to handle in the component logic (e.g., with the nullish coalescing operator ??).

Reply:
I personally prefer the empty of() because it is essentially an empty pipe. The default signal value will then be used because no change in the pipe will have happened. 

If we want to handle error messages later, we can change how it is handled.

Reply:
I have replaced it with EMPTY 👍